### ISSUE TITLE
EXC_BAD_ACCESS in Konan_ensureAcyclicAndSet in Lazy

### ISSUE THREAD

**Issue by benasher44** [JetBrains/kotlin-native#3856]
I noticed in our production crash reports we have a fair number of these kinds of crashes that show up accessing a `Lazy` var on a frozen object across multiple threads. Here's an example of what the setup looks like:

```kotlin
class Foo {}

class Holder {
    val foo: Foo by lazy { Foo() }
    init { freeze() }
}

```

Here's the K/N part of the trace:

```
EXC_BAD_ACCESS: Attempted to dereference garbage pointer 0xc0000000028.

0  PGFoundation      Konan_ensureAcyclicAndSet
1  PlanGrid          <anonymous> (Lazy.kt:40:26)
2  PlanGrid          getValue (Lazy.kt:44:93)
```

Let me know if I can provide any more info.

Kotlin 1.3.61, iOS 13.3, arm64

**Comment by olonho** [comment]
Would be amazing to have complete reproducer, if possible.

**Comment by benasher44** [comment]
I've tried putting together a reproducer, and so far I have been successful in reproducing. I've never seen it live myself, which is part of the problem. We only see it in our crash reporting tool.

**Comment by benasher44** [comment]
I DM'd you Kotlin slack some raw crash data

**Comment by SvyatoslavScherbina** [comment]
Fixed in 1.3.70.
